src/app/admin/generator/page.tsx

```
@@ -0,0 +1,409 @@
+                                <div className="space-y-2">
+                                    {tagGroupings.map(grouping => {
+                                        const isExpanded = expandedTagGroupings.has(grouping)
+                                        const groupingTags = tags.filter(tag => tag.grouping === grouping)
+                                        const selectedCount = groupingTags.filter(tag => selectedTags.includes(tag.name)).length
+
+                                        return (
+                                            <div key={grouping} className="border rounded-lg overflow-hidden bg-card">
+                                                <button
+                                                    onClick={() => toggleTagGrouping(grouping)}
+                                                    className="w-full px-4 py-3 flex items-center justify-between bg-muted/30 hover:bg-muted/50 transition-colors text-left"
+                                                >
```

Biome (2.3.13)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)